### ISSUE TITLE
Add vips support.

### ISSUE THREAD

**Issue by skunkworker** [refile/refile#197]
Hi, for a while I've been using the ruby-vips library to do my image resizing and cropping as it gives a substantial performance boost over using imagemagick. In some cases up to 4.2X faster while using 1/18th the memory. And on the projects that i've been working on that do a lot of resizing and cropping having this support is a deal breaker.
[http://www.vips.ecs.soton.ac.uk/index.php?title=Speed_and_Memory_Use](http://www.vips.ecs.soton.ac.uk/index.php?title=Speed_and_Memory_Use)
Now I could just fork the project but the main question is, would it be better to create an option that allows for vips to be turned on just for those specific operations or what is the best way to go about integrating this?
As an example in the past using carrierwave we had the carrierwave-vips gem that integrated nicely, but I still see some coupling in the project with imagemagick.

**Comment by janko-m** [comment]
This is definitely an interesting proposal. I heard about Vips as well. Originally after extracting the image processing from Refile into a separate gem, we intended to make MiniMagick a dependency of that gem. If we would support Vips, however, we could probably have a configuration switch on Refile which processing library to use. I did have problems figuring out Vips on the command line, does the ruby-vips library make things easier?

**Comment by skunkworker** [comment]
@janko-m I'm working on porting over the carrierwave-vips gem integration as it uses similar commands (limit, fit, fill, convert) on a branch that I forked off of the project. Most of the logic can be borrowed from that gem. The biggest downside to using vips instead of other libraries is that it really only supports PNG, JPEG & uncompressed formats. Gifs aren't supported. But then again on a lot of the projects that I have worked on this wasn't an issue and actually not allowing gifs was a good thing. Its a tradeoff for sure but the immediate performance improvements are substantial and greatly improve the ux of waiting for a photo to finish uploading and cropping etc.
